limitador/src/storage/distributed/mod.rs

```diff
@@ -279,6 +296,49 @@ impl CrInMemoryStorage {
     }
 }
 
+async fn process_re_sync(
+    limits: &Arc<RwLock<HashMap<Vec<u8>, CrCounterValue<String>>>>,
+    sender: Sender<Option<CounterUpdate>>,
+) {
+    // sending all the counters to the peer might take a while, so we don't want to lock
+    // the limits map for too long, lets figure first get the list of keys that needs to be sent.
+    let keys: Vec<_> = {
+        let limits = limits.read().unwrap();
+        limits.keys().cloned().collect()
+    };
+
+    for key in keys {
+        let update = {
+            let limits = limits.read().unwrap();
+            limits.get(&key).and_then(|store_value| {
+                let (expiry, ourself, value) = store_value.clone().into_ourselves_inner();
+                if value == 0 {
+                    None // no point in sending a counter that is empty
```

Comment:
Because, that's a send, you decided that each node would only send the counter that they have a value for "themselves" and let the others do the same for _their own values_? Am I now understanding this right?

Reply:
Correct.  But I think we should also check the expired condition, so I'll update that.